dpdata/rdkit/sanitize.py

```diff
@@ -0,0 +1,568 @@
+                        backward_btype = mol.GetBondBetweenAtoms(ring[i], ring[(i - 1) % lring]).GetBondType()
+                        if forward_btype != BondType.AROMATIC:
+                            switch = forward_btype == BondType.DOUBLE
+                            d1, e1 = search_and_assign_ring(mol, ring, hetero, i, forward=False, start_switch=switch)
+                            d2 = e2 = 0
+                        elif backward_btype != BondType.AROMATIC:
+                            switch = backward_btype == BondType.DOUBLE
+                            d1, e1 = search_and_assign_ring(mol, ring, hetero, i, forward=True, start_switch=switch)
+                            d2 = e2 = 0
+                        else:
+                            d1, e1 = search_and_assign_ring(mol, ring, hetero, i, forward=True, start_switch=True)
+                            d2, e2 = search_and_assign_ring(mol, ring, hetero, i, forward=False, start_switch=False)
+                        n_targetDouble -= (d1 + d2)
+                        n_targetEdit -= (e1 + e2)
+                    else:
+                        d1, e1 = search_and_assign_ring(mol, ring, hetero, i, forward=True, start_switch=True)
+                        d2, e2 = search_and_assign_ring(mol, ring, hetero, i, forward=False, start_switch=True)
+                        n_targetDouble -= (d1 + d2)
+                        n_targetEdit -= (e1 + e2)
+
+        for ring in CAr:
+            lring = len(ring)
+            for i in range(lring):
+                bond = mol.GetBondBetweenAtoms(ring[i], ring[(i + 1) % lring])
+                bond.SetBondType(BondType.AROMATIC)
+        print("Manual kekulization for aromatic heterocycles:")
+        print_bondtypes(mol, rings)
+
+        atoms = mol.GetAtoms()
+        for i in range(len(atoms)):
+            mol.ReplaceAtom(i, Chem.Atom(atoms[i].GetSymbol()))
+        mol_edited = mol.GetMol()
+        # charge assignment
+        if assign_formal_charge:
+            mol_edited = regularize_formal_charges(mol_edited, sanitize=False)
+        if not sanitize:
+            return mol_edited
+        else:
+            try:
+                Chem.SanitizeMol(mol_edited)
+                return mol_edited
+            except Exception as e:
+                raise RuntimeError(f"Manual kekulization for aromatic heterocycles failed, below are errors:\n\t {e}")
+
+def convert_by_obabel(mol, cache_dir=os.path.join(os.getcwd(), '.cache'), obabel_path="obabel"):
+    if not os.path.exists(cache_dir):
+        os.mkdir(cache_dir)
+    if mol.HasProp("_Name"):
+        name = mol.GetProp("_Name")
+    else:
+        name = f"mol{int(time.time())}"
+    mol_file_in = os.path.join(cache_dir, f'{name}.mol')
+    mol_file_out = os.path.join(cache_dir, f'{name}_obabel.mol')
+    Chem.MolToMolFile(mol, mol_file_in, kekulize=False)
+    out = os.popen(f"obabel {mol_file_in} -O {mol_file_out}").read()
```

Comment:
Can you consider to use openbabel's python API directly, which should have a better performance?
```suggestion
    from openbabel import openbabel
    obConversion = openbabel.OBConversion()
    obConversion.SetInAndOutFormats("mol", "mol")
    mol = openbabel.OBMol()
    obConversion.ReadFile(mol, mol_file_in)
    obConversion.WriteFile(mol, mol_file_out)
```